### ISSUE TITLE
hello, could you explain what's meaning of "with and without selected norm for other previous OOD detection methods"

### ISSUE THREAD

**Issue by yoksen** [gist-ailab/block-selection-for-OOD-detection#2]
I don't understand how to use the selected norm for other ood methods, could you please explain more details? For example, how to use selected norm in msp? Thanks.

**Comment by birdomi** [comment]
Hi, thank you for your question.

To provide a clear explanation, I will use ResNet18 as an example. The first step in logit calculation involves computing the dot product of the feature vector _f_ and the class weight vector _w_.

Then, we replace the magnitude of the feature vector _f_ with the magnitude of the output from the selected block, while keeping the direction of _f_ the same. We refer to this as "with our selected norm." After this step, the logit or probability (MSP) is calculated using the same process (e.g., softmax function).

Alternatively, when using logit "without the selected norm," the original feature vector and class weight vector are utilized to compute the logit. This produces the same results as using MSP or Energy method.

I hope that these comments have been helpful in providing a better understanding.
